src/meta/api/src/schema_api_impl.rs

```diff
@@ -145,6 +153,112 @@ const DEFAULT_DATA_RETENTION_SECONDS: i64 = 24 * 60 * 60;
 /// Thus every type that impl KVApi impls SchemaApi.
 #[tonic::async_trait]
 impl<KV: KVApi> SchemaApi for KV {
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn create_catalog(
+        &self,
+        req: CreateCatalogReq,
+    ) -> Result<CreateCatalogReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let name_key = &req.name_ident;
+        if req.meta.dropped_on.is_some() {
+            return Err(KVAppError::AppError(AppError::CreateCatalogWithDropTime(
+                CreateCatalogWithDropTime::new(&name_key.ctl_name),
+            )));
+        }
+
+        let mut retry = 0;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
+            let (ctl_id_seq, ctl_id) = get_u64_value(self, name_key).await?;
+            debug!(ctl_id_seq, ctl_id, ?name_key, "get_catalog");
+
+            if ctl_id_seq > 0 {
+                return if req.if_not_exists {
+                    Ok(CreateCatalogReply { ctl_id })
+                } else {
+                    Err(KVAppError::AppError(AppError::CatalogAlreadyExists(
+                        CatalogAlreadyExists::new(
+                            &name_key.ctl_name,
+                            format!("create catalog: tenant: {}", name_key.tenant),
+                        ),
+                    )))
+                };
+            }
+
+            // get ctl id list from _fd_ctl_id_list/ctl_id
+            let ctl_id_list_key = CatalogIdListKey {
+                tenant: name_key.tenant.clone(),
+                ctl_name: name_key.ctl_name.clone(),
+            };
```

Comment:
Are you going to support time travel for the catalog API?

Reply:
Nope